admin/src/main/resources/io/buoyant/admin/js/src/metrics_collector.js

```diff
@@ -7,19 +7,19 @@ define(['jQuery'], function($) {
 
   var MetricsCollector = (function() {
     var updateUri = "admin/metrics.json?tree=1";
-    var listeners = [];
+    var listeners = {};
     var clients = {};
 
     /**
       Register a listener to receive metric updates.
       handler: function called with incoming tree data
     */
-    function registerListener(handler) {
-      listeners.push({handler: handler});
+    function registerListener(listenerId, handler) {
+      listeners[listenerId] = handler;
     }
 
-    function deregisterListener(handler) {
-      _.remove(listeners, function(l) { return l.handler === handler; });
+    function deregisterListener(listenerId) {
+      delete(listeners[listenerId]);
```

Comment:
ah! I guess what I was saying was that since clients can now be expired and unexpired at any time, we'd call register/deregister more often than before, where we never deregistered anything unless the user minimized the client section